bpf/bpf_overlay.c

```diff
@@ -141,7 +141,7 @@ __section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_IPV4) int tail_handle_ipv4(struct _
 	int ret = handle_ipv4(skb);
 
 	if (IS_ERR(ret))
-		return send_drop_notify_error(skb, ret, TC_ACT_SHOT);
+		return send_drop_notify_error(skb, ret, TC_ACT_SHOT, DIRECTION_EGRESS); //Check if direction is correct
```

Comment:
The section includes `CILIUM_CALL_IPV4`. Tracing the `ep_tail_call(...)` call for this enum shows that it is in the section `from-overlay`. This suggests that it should actually be `DIRECTION_INGRESS`. Furthermore, tracing that function to `handle_ipv6` inside `bpf/bpf_overlay.c` shows calls to `skb_get_tunnel_key()`, ie it's fetching info from the tunnel metadata, and also `ipv6_local_delivery()`, so it's sending to the local stack, not into the tunnel. 

Similarly for the below call.

Reply:
thanks.. will change

Reply:
done